packages/react-components/src/components/MentionPopover.tsx

```diff
@@ -0,0 +1,332 @@
+      finalPosition.right = (rect?.width ?? 0) - leftOffset;
+    } else {
+      finalPosition.left = leftOffset;
+    }
+
+    if (location === 'below') {
+      finalPosition.top = (rect?.height ?? 0) + (targetPositionOffset?.top ?? 0);
+    } else {
+      // (location === 'above')
+      finalPosition.bottom = (rect?.height ?? 0) - (targetPositionOffset?.top ?? 0);
+    }
+    setPosition(finalPosition);
+  }, [location, target, targetPositionOffset]);
+
+  const handleOnKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      switch (e.key) {
+        case 'Escape':
+          onDismiss && onDismiss();
+          break;
+        default:
+          break;
+      }
+    },
+    [onDismiss]
+  );
+
+  const personaRenderer = useCallback(
+    (displayName?: string): JSX.Element => {
+      const displayNamePlaceholder = localeStrings.participantItem.displayNamePlaceholder;
+      const avatarOptions = {
+        text: displayName?.trim() || displayNamePlaceholder,
+        size: PersonaSize.size24,
+        initialsColor: theme.palette.neutralLight,
+        initialsTextColor: theme.palette.neutralSecondary,
+        showOverflowTooltip: false,
+        showUnknownPersonaCoin: !displayName?.trim() || displayName === displayNamePlaceholder
+      };
+
+      return <Persona {...avatarOptions} />;
+    },
+    [localeStrings, theme]
+  );
+
+  const defaultOnRenderSuggestionItem = useCallback(
+    (suggestion: Mention, onSuggestionSelected: (suggestion: Mention) => void, active: boolean): JSX.Element => {
+      return (
+        <div
+          data-is-focusable={true}
+          /* @conditional-compile-remove(mention) */
+          data-ui-id={ids.mentionSuggestionItem}
+          key={suggestion.id}
+          onClick={() => onSuggestionSelected(suggestion)}
+          onMouseEnter={() => setHoveredSuggestion(suggestion)}
+          onMouseLeave={() => setHoveredSuggestion(undefined)}
+          onKeyDown={(e) => {
+            handleOnKeyDown(e);
+          }}
+          className={suggestionItemWrapperStyle(theme)}
+        >
+          <Stack
+            horizontal
+            className={suggestionItemStackStyle(
+              theme,
+              hoveredSuggestion?.id === suggestion.id,
+              (changedSelection ?? false) && active
+            )}
+          >
+            {personaRenderer(suggestion.displayText)}
+          </Stack>
+        </div>
+      );
+    },
+    [
+      handleOnKeyDown,
+      theme,
+      /* @conditional-compile-remove(mention) */
+      ids,
+      hoveredSuggestion,
+      changedSelection,
+      personaRenderer
+    ]
+  );
+
+  const getHeaderTitle = useCallback((): string => {
+    if (title) {
+      return title;
+    }
+    /* @conditional-compile-remove(mention) */
+    return localeStrings.mentionPopover.mentionPopoverHeader;
+    return '';
+  }, [localeStrings, title]);
+
+  return (
+    <div ref={popoverRef}>
```

Comment:
BTW, a QQ for this part, why don't we use fluent ui component for this PopOver but implement our own div element instead, it feels like lots of logic we could reuse from fluent ui component

Reply:
Fluent 9 has a popover; I think we can move to it when we get to fluent 9

Reply:
Let's open an further task them, just in case we forget it someday when we get fluent 9